presto-tests/src/test/java/com/facebook/presto/execution/TestHistoryBasedStatsTracking.java

```diff
@@ -471,6 +474,23 @@ public void testBroadcastJoin()
                         anyTree(any())));
     }
 
+    @Test
+    public void testFactPrioritization()
+    {
+        String query1 = "SELECT (SELECT nationkey FROM nation WHERE name = 'UNITED STATES') AS us_nationkey";
+        executeAndTrackHistory(query1);
+        assertPlan(query1, anyTree(node(EnforceSingleRowNode.class, anyTree(any()))
+                .withOutputRowCount(1)
+                .withSourceInfoAndConfidenceLevel(new CostBasedSourceInfo(FACT))));
+
+        Session session = Session.builder(createSession()).setSystemProperty("prefer_partial_aggregation", "false").build();
```

Comment:
Why we need this session property?

Reply:
This session property prevent getting a aggregation remote hash exchange error, which was a fix suggested by Sreeni